src/app.ts

```diff
@@ -0,0 +1,22 @@
+import express, { Application, Request, Response, NextFunction } from 'express';
+const app: Application = express();
+const cors = require('cors');
+import postRoutes from './routes/posts' // bring in the routes to use here and then remove them below
+
+app.set('port', process.env.PORT || 5000);
+
+app.use(express.json());
+app.use(cors());
+
+// set up root route for posts
+app.use('/api/v1/posts', postRoutes)
+
+// do some basic error handling
+app.use((err: Error, req: Request, res: Response, next: NextFunction) => {
```

Comment:
`next` shouldn't be needed here, given that it's not being called. The method signature should allow for it to be completely omitted

Reply:
Admittedly, I don't really get what NextFunction is supposed to do.  Maybe it's a callback??

Reply:
Basically? It's like a... "fall through to the next handler" type of method... So like,

```ts
// Handler 1
app.use((..., next) => {
  if (canDoStuff) {
    // Do Stuff
  } else {
    next(); // Go to handler 2
  }
});

// Handler 2
app.use((...) => {
  // Now I'm here
});
```

Reply:
ahhhh. Sweet.